src/backend/distributed/deparser/ruleutils_18.c

```diff
@@ -1313,6 +1345,9 @@ set_relation_column_names(deparse_namespace *dpns, RangeTblEntry *rte,
 			has_anonymous = true;
 	}
 
+	/* We're now done needing the colinfo's names_hash */
```

Comment:
This seems to be put in the wrong place. Check the commit again - in front of it there is a line "changed_any = true;", not "has_anonymous = true;" 
https://github.com/postgres/postgres/commit/52c707483ce4d0161127e4958d981d1b5655865e#diff-6c12827097e95912476267d73f3f56e8250b13aad37ab4286c3c7fdab7f30ec7R4480

Reply:
You are right. Sorry about it.